### DISCUSSION TITLE
Meaning of each error code?

### DISCUSSION THREAD

**Discussion by pobch** [axios/axios#4971] [Q&A] [answered]
By scanning through the source code, there are 10 different error codes.
```
  static readonly ERR_FR_TOO_MANY_REDIRECTS = "ERR_FR_TOO_MANY_REDIRECTS";
  static readonly ERR_BAD_OPTION_VALUE = "ERR_BAD_OPTION_VALUE";
  static readonly ERR_BAD_OPTION = "ERR_BAD_OPTION";
  static readonly ERR_NETWORK = "ERR_NETWORK";
  static readonly ERR_DEPRECATED = "ERR_DEPRECATED";
  static readonly ERR_BAD_RESPONSE = "ERR_BAD_RESPONSE";
  static readonly ERR_BAD_REQUEST = "ERR_BAD_REQUEST";
  static readonly ERR_CANCELED = "ERR_CANCELED";
  static readonly ECONNABORTED = "ECONNABORTED";
  static readonly ETIMEDOUT = "ETIMEDOUT";
```
Could someone provide description of each error code? What is the condition causes each error? It does not seem to appear in the documentation.

**Comment by akbar-idwise** [comment]
I think  `ERR_CANCELED` is for when a user cancels the request using cancel token & `ECONNABORTED` is when you refresh the page when a request is on the fly or any similar condition.

**Comment by akbar-idwise** [comment]
Just asked this on [stack overflow](https://stackoverflow.com/questions/76288625/what-is-the-meaning-of-these-axios-error-codes)

**Comment by knulledge** [comment, marked as answer]
Please refer to [Error Types](https://github.com/axios/axios?tab=readme-ov-file#error-types) section of README
